File: Problem-1-CheckEven/solution.go
// For Explanation read Solution.java
package main  
 
import "fmt"

// Method 1
func usingBitwiseOper(n int) bool{
    return (n & 1) == 0;
}

// Method 2
func usingIntegerDivision(n int) bool{
    if((n / 2) * 2 == n) {
        return true;
    }
    return false;
}

func main() {

	fmt.Println(usingBitwiseOper(0));
	fmt.Println(usingBitwiseOper(1));
	fmt.Println(usingBitwiseOper(4));
	fmt.Println(usingBitwiseOper(99));

	// fmt.Println(usingIntegerDivision(0));
	// fmt.Println(usingIntegerDivision(1));
	// fmt.Println(usingIntegerDivision(4));
	// fmt.Println(usingIntegerDivision(99));
}

File: Problem-1-CheckEven/solution.py
# For Explanation read Solution.java

# Method 1
def usingBitwiseOper(n):
    return not bool(n & 1)


# Method 2
def usingIntegerDivision(n):
    if((n // 2) * 2 == n):
        return True
    return False


        
print(usingBitwiseOper(0))
print(usingBitwiseOper(1))
print(usingBitwiseOper(4))
print(usingBitwiseOper(99))

# print(usingIntegerDivision(0))
# print(usingIntegerDivision(1))
# print(usingIntegerDivision(4))
# print(usingIntegerDivision(99))
